surmise/calibration.py

```diff
@@ -259,11 +265,37 @@ def predict(self, x=None, args=None):
         if 'predict' in dir(self.method):
             self.method.predict(info, self.info, self.emu, x, args)
         else:
-            emupred = self.emu.predict(x, self.theta.rnd(1000))
+            nsamp = 1000
+            emupred = self.emu.predict(x, self.theta.rnd(nsamp))
             info['mean'] = np.mean(emupred.mean(), 1)
             info['var'] = np.var(emupred.mean(), 1)
             info['rnd'] = (emupred.mean()).T
-        return prediction(info, self)
+
+        predobj = prediction(info, self)
+        predobj.empirical_coverage()
+        return predobj
+
+    @staticmethod
+    def cast_f64_dtype(x):
+        return np.array(x, dtype=np.float64)
+
+    def save(self, filename):
+        """
+        Simple serialization and save function for calibrator object.
+
+        :Example:
+        >>> import dill
+        >>> cal = calibrator(...)
+        >>> # save calibrator object
+        >>> cal.save('cal_example.pkl')
+        >>> # load calibrator object
+        >>> with open('cal_example.pkl', 'rb') as f:
+        >>>     loaded_cal = dill.load(f)

```

Comment:
Your `save` function hides the fact that you are using `dill` to save your calibrator to file.  This is nice because you hide implementation details from the users behind `save`'s interface.  For instance, if you later decide to use a different tool to write/read, you can change the implementation of `save` and your users won't know that you have done that nor need to change their code.

However, you expose them to the use of `dill` when they load.  Could you add something like a `from_file` class method to you class?  Example:
```
cal = calibrator.from_file("my_cal.surmise")
```

Note that by suggesting an extension such as "surmise" you are also hiding the nature of the implementation.  Although that might be a step too far.

If it's important to your overall design that all surmise objects be written to file in the same way, then you might want to make a package-wide helper function that does the writing/reading.  Your class methods could then just call those.

Reply:
Resolved in 604be2d038044c41c43c47978e7f215fb44113dc